jwql/utils/utils.py

```diff
@@ -458,6 +479,59 @@ def filename_parser(filename):
         r"_(?P<instrument>(nircam|niriss|nirspec|miri|fgs))"\
         r"_(?P<optical_elements>((?!_)[\w-])+)"
 
+    # Stage 3 WFSS filenames
+    # e.g. jw01076-o124_s000001220_nircam_f322w2-grismr_x1d.fits cal.fits c1d.fits
+    #This should not be used anymore...
+    stage_3_wfss_source_id = \
```

Comment:
Wow -- this is impressive. Lot's of regex here!

Reply:
So many file naming options to cover....